### PR TITLE
Add getFragDataLocation to gl.object-get-call.js

### PR DESCRIPTION
**Pull request by renchenglei**
Please help review the code, thanks a lot~

### CODE REVIEW

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 1]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
 	    return gl.getQueryParameter(query, pname);
 	}
     );
+
+    debug("");
+    debug("Test getFragDataLocation");
+    var baseVertShader = '' +
+    '#version 300 es\n' +
+    'uniform vec4 normal;\n' +
+    'in vec4 vertext;\n' +
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: vertext -> vertex

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 2]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
...
+    var fragShader = gl.createShader(gl.FRAGMENT_SHADER);
+    gl.shaderSource(fragShader, baseFragShader);
+    gl.compileShader(fragShader);
+    shouldBe('gl.getShaderParameter(fragShader, gl.COMPILE_STATUS);', 'true');
+    var program = gl.createProgram();
+    gl.attachShader(program, vertShader);
+    gl.attachShader(program, fragShader);
+    gl.linkProgram(program);
+    shouldBe('gl.getProgramParameter(program, gl.LINK_STATUS)','true');
+    shouldBe('gl.getFragDataLocation(program, "vertextColor")', '-1');
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: vertextColor -> vertexColor

File: sdk/tests/js/tests/gl-object-get-calls.js (hunk 1 above)
Comment (renchenglei) [inline sdk/tests/js/tests/gl-object-get-calls.js]: thanks a lot!

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 3]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
 	    return gl.getQueryParameter(query, pname);
 	}
     );
+
+    debug("");
+    debug("Test getFragDataLocation");
+    var baseVertShader = '' +
+    '#version 300 es\n' +
+    'uniform vec4 normal;\n' +
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: normal's unused. Either reference it in the shader so it isn't optimized away or remove it.

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 4]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
...
+    'in vec4 vertext;\n' +
+    'out vec4 position;\n' +
+    'void main (void)\n' +
+    '{\n' +
+    '       position = vec4(0.0);\n' +
+    '       gl_Position = vertext;\n' +
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: Assignment to gl_Position seems useless. Just assign position to vertex?

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 5]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
...
+    gl.compileShader(fragShader);
+    shouldBe('gl.getShaderParameter(fragShader, gl.COMPILE_STATUS);', 'true');
+    var program = gl.createProgram();
+    gl.attachShader(program, vertShader);
+    gl.attachShader(program, fragShader);
+    gl.linkProgram(program);
+    shouldBe('gl.getProgramParameter(program, gl.LINK_STATUS)','true');
+    shouldBe('gl.getFragDataLocation(program, "vertextColor")', '-1');
+    shouldBe('gl.getFragDataLocation(program, "normal;")', '-1');
+    shouldBeNonZero('gl.getFragDataLocation(program, "vertext")');
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: Why is this the case? "vertex" is not a varying out fragment shader variable.

File: sdk/tests/js/tests/gl-object-get-calls.js [hunk 6]
```diff
@@ -865,7 +865,47 @@ if (contextVersion > 1) {
...
+    shouldBe('gl.getFragDataLocation(program, "position")', '0');
```
Comment (kenrussell) [inline sdk/tests/js/tests/gl-object-get-calls.js]: Why is this the case? position is an input variable to the fragment shader, not an output variable. It should be -1, no?

**Comment by kenrussell** [comment]
Thanks for this contribution. Looks like there are a few issues.

File: sdk/tests/js/tests/gl-object-get-calls.js (hunk 5 above)
Comment (renchenglei) [inline sdk/tests/js/tests/gl-object-get-calls.js]: So sorry, it should be normal.

**Comment by renchenglei** [comment]
Thanks for your advice. I will modify as soon as possible.
@kenrussell

**Comment by renchenglei** [comment]
Please help review the code, thanks a lot~
@kenrussell

**Comment by kenrussell** [comment]
Thanks for updating this test. It looks good.

**Pull request merged**